mne/viz/raw.py

```diff
@@ -566,31 +566,59 @@ def _set_psd_plot_params(info, proj, picks, ax, area_mode):
             ax_list, make_label)
 
 
-def _convert_psds(psds, dB, scaling, unit, ch_names):
-    """Convert PSDs to dB (if necessary) and appropriate units."""
-    if dB:
-        where = np.where(psds.min(1) <= 0)[0]
-        if len(where) > 0:
-            raise ValueError("Infinite value in PSD for channel(s) %s. "
-                             "These channels might be dead." %
-                             ', '.join(ch_names[ii] for ii in where))
+def _convert_psds(psds, dB, estimate, scaling, unit, ch_names):
+    """Convert PSDs to dB (if necessary) and appropriate units.
+
+    The following table summarizes the relationship between the value of
+    parameters `dB` and `estimate`, and the type of plot and corresponding
+    units.
+
+    | dB    | estimate    | plot | units           |
+    |-------+-------------+------+-----------------|
+    | True  | 'power'     | PSD  | u**2/Hz (dB)    |
+    | False | 'amplitude' | ASD  | u/sqrt{Hz}      |
+    | True  | 'auto'      | PSD  | u**2/Hz (dB)    |
+    | False | 'power'     | PSD  | u**2/Hz         |
+    | True  | 'amplitude' | ASD  | u/sqrt{Hz} (dB) |
+    | False | 'auto'      | ASD  | u/sqrt{Hz}      |
+    """
+
+    where = np.where(psds.min(1) <= 0)[0]
+    if len(where) > 0:
+        raise ValueError("Infinite value in PSD for channel(s) %s. "
```

Comment:
If `dB=False` the power is not infinite but zero.

Reply:
OK. I wonder if this should be a warning rather than an Exception (I imagine someone might have a channel short-circuited on purpose.

Reply:
+1 for warning

Reply:
fine by me

Reply:
Throwing a warning opens a new can of worms. Now when computing the log10, I get NumPy warnings. Do we just let them there? They look kind of ugly. But disabling these warnings in the code is not nice either.

Reply:
`np.log10(np.maximum(whatever, np.finfo(float).tiny))`

Reply:
Thanks. Just pushed with this suggestion.